src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/DecimalMathFuncs.txt

```diff
@@ -0,0 +1,453 @@
+
+>> RoundUp( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(300000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:300000.13})
+
+>> RoundDown( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(30000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:30000.12})
+
+// these rounding tests are returning the correct results, due to the conversion between decimal and float
+
+>> Int( [Float(100000.12), 1000000000000000000000001.1, 2000000000000000000000002.2] )
+Table({Value:100000},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> Trunc( [Float(100000.12), 1000000000000000000000001.1, 2000000000000000000000002.2] )
+Table({Value:100000},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> Round( [Float(300000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:300000.12},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> RoundUp( [Float(300000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:300000.13},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> RoundDown( [Float(30000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:30000.12},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+// non-decimal aware numeric functions
+
+>> Sqrt( 1524157899404237160815376 )         // exact square of 1234567899876, should be done in float and match next test
+1234567899876
+
+>> Sqrt( Float(1524157899404237160815376) )  // exact square of 1234567899876
+1234567899876
+
+>> Sqrt( 4.000000000000000004 )              // outside of float range, will round to 4
+2
+
+>> Sqrt( 4.0000000004 )                      // within float range
+2.0000000001
+
+>> Log( 100, 10 )
+2
+
+>> Log( 10000000000000050000000, 10 )         // input will be rounded with float conversion
+22
+
+>> Log( 10000000000000050000000, Float(10) )
+22
+
+>> Log( 10000000000000000000000, 10 )         // above tests should match these answers
+22
+
+>> Log( 10000000000000000000000, Float(10) )  
+22
+
+>> Log( Float("1e100"), 10 )
+100
+
+>> Ln( 2.71828182845904523536028747135266249775724709369995 )
+1
+
+>> Ln( 100000000 )
+18.420680743952367
+
+>> Exp( 18.420680743952367 )
+100000000.00000007
+
+>> Pi()
+3.141592653589793
+
+>> Cos( 0 )
+1
+
+>> Cos( [0] )
+Table({Value:1})
+
+>> Cos( Pi() )
+-1
+
+>> Cos( [Pi()] )
+Table({Value:-1})
+
+>> Sin( 0 )
+0
+
+>> Sin( [0] )
+Table({Value:0})
+
+>> Sin( Pi() )
+1.2246467991473532E-16
+
+>> Cot( 0 )
+Error({Kind:ErrorKind.Div0})
+
+>> Cot( [0] )
+Table({Value:Error({Kind:ErrorKind.Div0})})
+
+>> Cot( 3.14159265358979323846264338327950288419716939937510 )
+-8165619676597685
+
+>> Cot( [3.14159265358979323846264338327950288419716939937510] )
+Table({Value:-8165619676597685})
+
+>> Cot( Pi() )
+-8165619676597685
```

Comment:
What's the value of this test which is wrong?
It should return an overflow error

Reply:
Good call!  Fixed.